packages/test/test-end-to-end-tests/src/test/sharedInterval.spec.ts

```diff
@@ -776,7 +776,7 @@ describeFullCompat("SharedInterval", (getTestObjectProvider) => {
             assert.equal(
                 mapFrom3.get("nestedKey"), "nestedValue", "Incorrect value in interval collection's shared map");
 
-            const summaryBlob = outerString2.summarize().summary.tree.header as ISummaryBlob;
+            const summaryBlob = (await outerString2.summarize()).summary.tree.header as ISummaryBlob;
```

Comment:
I'm a bit concerned that the test fails with the following error if it has `outerString2.getAttachSummary` instead. Compat issue?

outerString2.getAttachSummary is not a function
TypeError: outerString2.getAttachSummary is not a function
    at Context.<anonymous> (dist/test/sharedInterval.spec.js:616:46)

Reply:
This code is wrapped with describeFullCompat() which will try to permutate various versions of container runtime & data stores. You are changing interfaces that are not on compat boundary, but this test code assumes "latest and greatest" by using getAttachSummary() as this API did not exist in previous version of data store (/ dds). But describeFullCompat() will given you a provider that will load old code (old data store code and thus old DDSs) that do not have getAttachSummary().

In other words, there is incompatibility of this test code (assuming latest) with actual data store you are getting.
You would need to either use methods that exists in both (that's what you do with summarize()) or disabled back-compat testing for a while.
You will be able to start leveraging getAttachSummary() in two releases.